src/main/scala/inox/solvers/princess/PrincessSolver.scala

```diff
@@ -36,6 +36,7 @@ trait PrincessSolver extends AbstractUnrollingSolver { self =>
   object templates extends {
     val program: targetProgram.type = targetProgram
     val context = self.context
+    protected implicit val semantics: targetProgram.Semantics = self.targetSemantics
```

Comment:
Nit: no need to make the override val implicit here.
Nor protected probably.

Reply:
Indeed, fixed!